File: src/main/java/com/curie/curieapp/entities/Profissao.java
package com.curie.curieapp.entities;

import com.curie.curieapp.entities.enums.Demanda;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "profissoes")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Profissao {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false)
    private Integer id;

    @Column(nullable = false)
    private String nome;

    @Column(nullable = false)
    private String descricao;

    @Column(nullable = false)
    private Double salario;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Demanda demanda;

}

File: app/src/main/java/com/curie/curie/data/model/Profissao.kt
package com.curie.curie.data.model

import com.curie.curie.data.model.enums.Demanda

data class Profissao(
    val id: Long,
    val nome: String,
    val descricao: String,
    val salario: Double,
    val demanda: Demanda
)
